tests/test_kernel_flexible_centerer.py

```diff
@@ -5,6 +5,36 @@
 
 
 class KernelTests(unittest.TestCase):
+    def test_sample_weights(self):
+        """Checks that sample weights of one are equal to the unweighted case and that nonuniform weights are different from the unweighted case"""
+        K = np.random.uniform(0, 100, size=(3, 3))
+        uniform_wts = np.ones(len(K))
+        nonuniform_wts = np.random.uniform(0, 100, size=(len(K),))
+        model = KernelFlexibleCenterer()
+        weighted_model = KernelFlexibleCenterer()
+        K_unweighted = model.fit_transform(K)
+        K_uniform_weighted = weighted_model.fit_transform(K, sample_weight=uniform_wts)
+        self.assertTrue(
+            (np.isclose(K_unweighted, K_uniform_weighted, atol=1e-12)).all()
+        )
+        K_nonuniform_weighted = weighted_model.fit_transform(
+            K, sample_weight=nonuniform_wts
+        )
+        self.assertFalse(
+            (np.isclose(K_unweighted, K_nonuniform_weighted, atol=1e-12)).all()
+        )
```

Comment:
Superficial comment here -- can you name this something other than `nonuniform`? It's a bit jarring to have it named `nonuniform` and then `np.random.uniform`. I would more suggest:
```suggestion
        equal_wts = np.ones(len(K))
        unequal_wts = np.random.uniform(0, 100, size=(len(K),))
        
        model = KernelFlexibleCenterer()
        weighted_model = KernelFlexibleCenterer()
        
        K_wo_weights = model.fit_transform(K)
        K_equally_weighted = weighted_model.fit_transform(K, sample_weight=equal_wts)
        self.assertTrue(
            (np.isclose(K_wo_weights, K_equally_weighted, atol=1e-12)).all()
        )
        K_randomly_weighted = weighted_model.fit_transform(
            K, sample_weight=random_wts
        )
        self.assertFalse(
            (np.isclose(K_wo_weights, K_randomly_weighted, atol=1e-12)).all()
        )
```

Reply:
I have changed all of the "uniform" to "equal" and "nonuniform" to "nonequal"